powerfactory/powerfactory-dgs/src/main/java/com/powsybl/powerfactory/dgs/DgsParser.java

```diff
@@ -40,62 +68,152 @@ private static void readTableHeader(String trimmedLine, DgsHandler handler, Pars
         String className = fields[0].substring(2);
         handler.onTableHeader(className);
 
-        context.attributesName = new String[fields.length - 1];
-        context.attributesType = new char[fields.length - 1];
-
-        for (int i = 1; i < fields.length; i++) {
-            String field = fields[i];
+        context.elementData = new ArrayList<>();
 
-            Matcher matcher = ATTR_DESCR_PATTERN.matcher(field);
-            if (!matcher.matches()) {
-                throw new PowerFactoryException("Invalid attribute description: '" + field + "'");
+        int index = 1;
+        while (index < fields.length) {
+            if (isMatrixHeader(fields, index, fields.length)) {
+                index = readMatrixHeader(fields, index, fields.length, handler, context);
+            } else if (isVectorHeader(fields, index)) {
+                index = readVectorHeader(fields, index, fields.length, handler, context);
+            } else {
+                readSingleHeader(fields, index, handler, context);
             }
+            index++;
+        }
+    }
 
-            String attributeName = matcher.group(1);
-            char attributeType = matcher.group(2).charAt(0);
-            handler.onAttributeDescription(attributeName, attributeType);
+    private static int readSingleHeader(String[] fields, int index, DgsHandler handler, ParsingContext context) {
```

Comment:
Maybe rename this method to `readAttributeHeader`?

Reply:
Done